File: JavaScript/BalancedBrackets.js
// Given n strings of brackets, determine whether each sequence of brackets is balanced. If a string is balanced, return YES. Otherwise, return NO.

function isBalanced(s) {

    var braks = []

    for (let i =0; i < s.length; i++) {
        if (s[i] == '(' || s[i] == '[' || s[i] == '{') {
            braks.push(s[i])
        } else if (s[i] == ']') {
            const possBracket = braks.pop()
            if (possBracket != '[') {
                return "NO"
            } 
        } else if (s[i] == ')') {
            const possBracket = braks.pop()
            if (possBracket != '(') {
                return "NO"
            } 
        } else if (s[i] == '}') {
            const possBracket = braks.pop()
            if (possBracket != '{') {
                return "NO"
            } 
        }
    };

    return "YES"
}

export { isBalanced }

File: Python/BalancedBrackets.py
# // Given n strings of brackets, determine whether each sequence of brackets is balanced. If a string is balanced, return YES. Otherwise, return NO.

def isBalanced(s):
    braks = []
    for i in range(len(s)):
        if (s[i] == '(' or s[i] == '[' or s[i] == '{'):
            braks.append(s[i])
        elif (s[i] == ']'):
            possBracket = braks.pop()
            if (possBracket != '['):
                return "NO"
            
        elif (s[i] == ')'):
            possBracket = braks.pop()
            if (possBracket != '('):
                return "NO"

        elif (s[i] == '}'):
            possBracket = braks.pop()
            if (possBracket != '{'):
                return "NO"

    return "YES"